### PR TITLE
Add ncc-gh action

### PR DESCRIPTION
**Pull request by Purushothaman22**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,25 @@
...
+
+## Inputs
+
+### `TOKEN`
+
+The GitHub token used to authenticate with GitHub.
+
+**Required**
+
+### `MAIN_FILLE_PATH`
```
Comment (gps) [inline README.md]: Typo

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+description:  '\"Builds\" a JS GitHub Action into a `dist/index.js` using @vercel/ncc'
+author: 'Gopal Sharma <gps@surya-soft.com>'
+inputs:
+  MAIN_FILE_PATH:
+    description: >
+      Path to main js file which will be used to generated `dist/index.js`.
+      Default value is `./index.js`.
+    required: false
+  TOKEN:
+    description: Auth token with delete permission.
```
Comment (gps) [inline action.yml]: Why do you need delete permission?

File: index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+async function run() {
+    const token = core.getInput('TOKEN');
+    var mainFilePath = core.getInput('MAIN_FILE_PATH')
+
+    if (!mainFilePath) {
+        mainFilePath = './index.js'
+    }
+
+    exec.exec('npm install');
+    exec.exec('npm i -g @vercel/ncc');
```
Comment (gps) [inline index.js]: Do we need the `-g`? Can't we do a local install?

**Review by gps — CHANGES_REQUESTED** [review]
Here's a weird question - can we set this action up on itself after the first merge to master? So that it builds itself using itself?

File: action.yml (hunk 2 above)
Comment (Purushothaman22) [reply action.yml]: Typo I had copy pasted from previous action

File: index.js (hunk 3 above)
Comment (Purushothaman22) [reply index.js]: Not required. We can do local install.

**Comment by Purushothaman22** [comment]
> Here's a weird question - can we set this action up on itself after the first merge to master? So that it builds itself using itself?

Even I thought about this, I'll look into it if it is possible, after I get this to working. Currently still making changes based on testing.

**Comment by Purushothaman22** [comment]
> Here's a weird question - can we set this action up on itself after the first merge to master? So that it builds itself using itself?

I've enabled this. Tested it with my private repository too.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+name: Buil Package to use ncc
```
Comment (gps) [inline .github/workflows/release.yml]: Typo

File: index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        const diff = await exec.exec(
+            'git', ['diff', '--quiet'], {ignoreReturnCode: true}
+        );
+    
+        if (diff || !distFolderAlreadyExists) {
+            await core.group('push changes', async () => {
+                await git.addConfig('user.email', `${env.GITHUB_ACTOR}@users.noreply.github.com`)
+                await git.addConfig('user.name', env.GITHUB_ACTOR)
+                await git.add('./dist')
+                await git.commit("Use  @vercel/ncc")
```
Comment (gps) [inline index.js]: Can you change this to: "Generate distribution"?

**Review by gps — CHANGES_REQUESTED** [review]

**Review by gps — APPROVED** [review]

**Pull request merged**